lib/evss/letters/service.rb

```diff
@@ -34,22 +34,38 @@ def get_letter_beneficiary(user)
       def download_by_type(user, type, options = nil)
         with_exception_handling do
           headers = headers_for_user(user)
-          if options.blank?
-            response = download_conn.get type do |request|
-              request.headers.update(headers)
-            end
+          response = make_download_request(headers, options, type)
+
+          case response.status.to_i
+          when 200
+            response.body
+          when 404
+            raise Common::Exceptions::RecordNotFound, type
           else
-            headers['Content-Type'] = 'application/json'
-            response = download_conn.post do |request|
-              request.url "#{type}/generate"
-              request.headers.update(headers)
-              request.body = options
-            end
+            log_message_to_sentry(
+              'EVSS letter generation failed', :error, extra_context: {
+                url: config.base_path, body: response.body
+              }
+            )
+            raise_backend_exception('EVSS502', 'Letters')
```

Comment:
I _think_ this will log to Sentry twice.  Once on your `log_message_to_sentry()` then again when `raise_backend_exception` raises `Common::Exceptions::BackendServiceException` which gets caught at the [`application_controller` rescue block](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/controllers/application_controller.rb#L46).  Probably OK to leave it in for now but keep an eye on Sentry and remove lines 44 through 49 if it makes too much noise.